src/renderers/WebGLRenderer.js

```diff
@@ -3017,47 +3021,96 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 	}
 
-	function setCubeTextureDynamic ( texture, slot ) {
+	function setTextureCubeDynamic ( texture, slot ) {
 
 		state.activeTexture( _gl.TEXTURE0 + slot );
 		state.bindTexture( _gl.TEXTURE_CUBE_MAP, properties.get( texture ).__webglTexture );
 
 	}
 
-	var setTextureWarned = false;
-	this.setTexture = function( texture, slot ) {
+	this.allocTextureUnit = allocTextureUnit;
 
-		if ( ! setTextureWarned ) {
+	this.setTexture2D = ( function() {
 
-			console.warn( "THREE.WebGLRenderer: .setTexture is deprecated, " +
-				"use setTexture2D instead." );
-			setTextureWarned = true;
+		var warned = false;
 
-		}
+		return function( texture, slot ) {
 
-		setTexture2D( texture, slot );
+			if ( texture instanceof THREE.WebGLRenderTarget ) {
 
-	};
+				if ( ! warned ) {
 
-	this.allocTextureUnit = allocTextureUnit;
-	this.setTexture2D = setTexture2D;
-	this.setTextureCube = function( texture, slot ) {
+					console.warn( "THREE.WebGLRenderer.setTexture2D: don't use render targets as textures. Use their .texture property instead." );
+					warned = true;
 
-		if ( texture instanceof THREE.CubeTexture ||
-			 ( Array.isArray( texture.image ) && texture.image.length === 6 ) ) {
+				}
 
-			// CompressedTexture can have Array in image :/
+				texture = texture.texture;
 
-			setCubeTexture( texture, slot );
+			}
 
-		} else {
-			// assumed: texture instanceof THREE.WebGLRenderTargetCube
+			setTexture2D( texture, slot );
 
-			setCubeTextureDynamic( texture.texture, slot );
+		};
 
-		}
+	}() );
 
-	};
+	this.setTexture = ( function() {
+
+		var warned = false;
+
+		return function( texture, slot ) {
+
+			if ( ! warned ) {
+
+				console.warn( "THREE.WebGLRenderer: .setTexture is deprecated, use setTexture2D instead." );
+				warned = true;
+
+			}
+
+			_this.setTexture2D( texture, slot );
+
+		};
+
+	}() );
+
+	this.setTextureCube = ( function() {
+
+		var warned = false;
+
+		return function( texture, slot ) {
+
+			if ( texture instanceof THREE.WebGLRenderTargetCube ) {
+
+				if ( ! warned ) {
+
+					console.warn( "THREE.WebGLRenderer.setTextureCube: don't use cube render targets as textures. Use their .texture property instead." );
+					warned = true;
+
+				}
+
+				texture = texture.texture;
+
+			}
+
+			if ( texture instanceof THREE.CubeTexture ||
+				 ( Array.isArray( texture.image ) && texture.image.length === 6 ) ) {
+
+				// CompressedTexture can have Array in image :/
+
+				setTextureCube( texture, slot );
+
+			} else {
+
+				// assumed: texture property of THREE.WebGLRenderTargetCube
```

Comment:
No it was `// assumed: texture instanceof THREE.WebGLRenderTargetCube` before and it's now assumed to be a property of a cube render target :(